internal/pgmonitor/exporter.go

```diff
@@ -152,12 +152,9 @@ func ExporterStartCommand(builtinCollectors bool, commandFlags ...string) []stri
 		`  echo "Starting postgres_exporter with the following flags..."`,
 		`  echo "${postgres_exporter_flags[@]}"`,
 		`  postgres_exporter "${postgres_exporter_flags[@]}" &`,
-		`  echo $! > $POSTGRES_EXPORTER_PIDFILE`,
+		`  [ -e /proc/$! ] && echo $! > $POSTGRES_EXPORTER_PIDFILE`,
```

Comment:
🤔 What effect does this change have?

Reply:
Only writing the pidfile if the proc exists? It was something that came up as leading to the flakes we saw. Not sure if we need it with the below changes.

Reply:
🔧 The command above this succeeded, so the process in `$!` very likely exists. I think it's best to always write the PID file so that later commands can always read it.